components/common/Link.tsx

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import * as React from "react";
+
+export const defaultAllowedSchemes = ['http:', 'https:']
+
+type Props = React.DetailedHTMLProps<React.AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement> & {
+  allowedSchemes?: string[]
+}
+
+export const validateScheme = (href: string | undefined, allowedSchemes: string[] = defaultAllowedSchemes) => {
+  // Special case the empty url - it shouldn't crash anything
+  if (!href) return
+
+  const url = new URL(href)
+  if (!allowedSchemes.includes(url.protocol)) {
+    throw new Error(`Unsupported scheme ${url.protocol} on url ${href}`)
+  }
+}
+
+/**
+ * Identical to the `a` element, except that it validates the HREF points
+ * to an allowed scheme. By default, allowed schemes are http: or https:,
+ * but this can be overridden.
+ */
+export default function Link({ href, allowedSchemes, ...rest }: Props) {
```

Comment:
Let's preface with Secure - SecureLink, SecureAnchor, SecureA - up to you. Just to make it clear what it's doing and it's not a styled element

Reply:
Good call! I've gone with the boring but obvious `SecureLink` even though `SecureA` (securer) makes me chuckle.